crates/component/src/wrappers/http.rs

```diff
@@ -0,0 +1,570 @@
+//! This module provides utilities for writing HTTP servers and clients using the WASI HTTP API.
+//!
+//! It's inspired by the WASI 0.3 proposal for <https://github.com/WebAssembly/wasi-http> and will
+//! be supported until the release of wasi:http@0.3.0. After that, this module will likely be deprecated.
+//!
+//! ```rust
+//! use wasmcloud_component::http;
+//!
+//! struct Component;
+//!
+//! http::export!(Component);
+//!
+//! // Implementing the [`Server`] trait for a component
+//! impl http::Server for Component {
+//!     fn handle(_request: http::IncomingRequest) -> http::Result<http::Response<impl http::OutgoingBody>> {
+//!         Ok(http::Response::new("Hello from Rust!"))
+//!     }
+//! }
```

Comment:
```suggestion
//! impl http::Server for Component {
//!     fn handle(_request: http::IncomingRequest) -> Result<Response<impl OutgoingBody>, ErrorCode> {
//!         Ok(http::Response::new("Hello from Rust!"))
//!     }
//! }
```

This currently matches the trait, but I think that the doc comment here indicates the desired behavior which is to return the `http::Response` instead of a `std::result::Result<http::Response<>, ErrorCode>`

Basically, I think this can stay the same if we just change the trait

Reply:
@rvolosatovs looking at the trait, I see
```rust
fn handle(request: IncomingRequest) -> Result<Response<impl OutgoingBody>, ErrorCode>;
```

But the doc comment suggests the trait is:

```rust
fn handle(request: IncomingRequest) -> http::Response<impl OutgoingBody>;
```

so I was just suggesting changing the trait response type to match the doc comment

Reply:
It's `http::Result<http::Response<impl http::OutgoingBody>>` in the doc actually, where `http::Result` is exported by this crate

Reply:
Oh! I see. Do you think we should change the `http::Result` type to be the following for compatibility? e.g. `builder.body(())` returns the `::http::Result<>` type.
```
pub type Result<T, E = http::Error> = core::result::Result<T, E>;
```

Reply:
> because [http::Error](https://docs.rs/http/latest/http/struct.Error.html) cannot be constructed from outside the http crate

Welp, that would make this more difficult then 😓 Resolving this then, I think we're in a good place then!